.github/workflows/build.yml

```diff
@@ -19,61 +18,65 @@ concurrency:
 jobs:
   test:
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
```

Comment:
Why this? The workflow has full access again after this.

Reply:
No, I have set "Read repository contents permission" for "Workflow permissions" in the repository settings instead of the default "Read and write permissions". Public repositories do not require it explicitly otherwise

Reply:
the above will only allow read repo contents, if it's removed, the workflow can also access settings / jobs / ...

Reply:
Again, I assure you the repository is configure with least-privileges by default.
Please see "Default access (restricted)" columns in this table: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token